README.md

```diff
@@ -33,15 +33,81 @@ val metadata: IO[RecordMetadata] = producer.use { producer =>
 ## Consumer usage example
 
 ```scala
+import cats.data.{NonEmptySet => Nes}
+
 val consumer = Consumer.of[IO, String, String](config, ecBlocking)
 val records: IO[ConsumerRecords[String, String]] = consumer.use { consumer => 
   for {
-    _       <- consumer.subscribe(Nel("topic"), None)
+    _       <- consumer.subscribe(Nes("topic"))
     records <- consumer.poll(100.millis)
   } yield records 
 }
 ```
 
+## Handling consumer group rebalance
+It's possible to provide a callback for a consumer group rebalance event, which can be useful if you want to do some computations,
+save the state, commit some offsets (or do anything with the consumer before the new group is applied).  
+This can be done by providing an implementation of `RebalanceListener1` (or a more convenient version - `RebalanceListener1WithConsumer`) 
+which requires you to return a `RebalanceCallback` structure which describes what actions should be performed in a certain situation. 
+It allows you to use some of consumer's methods as well as a way to run an arbitrary computation.
+Please note that all the actions are executed on the consumer `poll` thread which means that running heavy or 
+long-running computations is discouraged. 
```

Comment:
`heavy or long-running computations is discouraged.`
I think we need to add some short explanation why is it discouraged and what are the options if it's really really neded :) 

- discouraged coz if it takes too long then kafka consumer instance can be removed from consumer group (default is 5 minutes and then consumer instance considered dead roughly speaking), but most probably before those 5 default minutes we would fail with timeout exception (default is 1 minute there) in `ToTry`
- so to workaround those limitations user can adjust ToTry/max.poll.interval.ms timeouts

I realise now it's really hard to explain such important mechanics in fewer words, so mb it deservs a link to a dedicated section about ToTry/max.poll.interval.ms timeouts?

More details are available in following issues:
- https://github.com/evolution-gaming/skafka/issues/126
- https://github.com/evolution-gaming/skafka/issues/125

Reply:
Added some clarifications - please check now, will that be enough?